Userland/Libraries/LibGfx/CharacterBitmap.h

```diff
@@ -9,26 +10,30 @@
 #include "Size.h"
 #include <AK/RefCounted.h>
 #include <AK/RefPtr.h>
+#include <AK/StringView.h>
 
 namespace Gfx {
 
-class CharacterBitmap : public RefCounted<CharacterBitmap> {
+class CharacterBitmap {
 public:
-    static NonnullRefPtr<CharacterBitmap> create_from_ascii(const char* asciiData, unsigned width, unsigned height);
-    ~CharacterBitmap();
+    CharacterBitmap() = delete;
+    constexpr CharacterBitmap(StringView ascii_data, unsigned width, unsigned height)
+        : m_bits { ascii_data }
+        , m_size { width, height }
+    {
+    }
 
-    bool bit_at(unsigned x, unsigned y) const { return m_bits[y * width() + x] == '#'; }
-    const char* bits() const { return m_bits; }
+    constexpr ~CharacterBitmap() = default;
 
-    IntSize size() const { return m_size; }
-    unsigned width() const { return m_size.width(); }
-    unsigned height() const { return m_size.height(); }
+    constexpr bool bit_at(unsigned x, unsigned y) const { return m_bits[y * width() + x] == '#'; }
+    constexpr StringView bits() const { return m_bits; }
 
-private:
-    CharacterBitmap(const char* b, unsigned w, unsigned h);
+    constexpr IntSize size() const { return m_size; }
+    constexpr unsigned width() const { return m_size.width(); }
+    constexpr unsigned height() const { return m_size.height(); }
 
-    const char* m_bits { nullptr };
-    IntSize m_size;
+private:
+    StringView m_bits {};
```

Comment:
Do we actually want to have a StringView here? What if we don't constexpr-initialize the bitmap and create it on the fly, are we forced to keep a String alive alongside it?

Reply:
This is no change from the previous behavior using a `const char*`. The string is required to stay alive in the previous code also. This has nothing to do with it being initialized at compile-time.